src/Domain/ElementMap.hpp

```diff
@@ -0,0 +1,134 @@
+      noexcept {
+    return *block_map_;
+  }
+
+  const ElementId<Dim>& element_id() const noexcept { return element_id_; }
+
+  template <typename T>
+  tnsr::I<T, Dim, TargetFrame> operator()(
+      tnsr::I<T, Dim, Frame::Logical> source_point) const noexcept {
+    // Apply the affine map to the points
+    for (size_t d = 0; d < Dim; ++d) {
+      source_point.get(d) = source_point.get(d) * gsl::at(map_slope_, d) +
+                            gsl::at(map_offset_, d);
+    }
+    return block_map_->operator()(source_point);
+  }
+
+  template <typename T>
+  tnsr::I<T, Dim, Frame::Logical> inverse(
+      const tnsr::I<T, Dim, TargetFrame>& target_point) const noexcept {
+    auto source_point{block_map_->inverse(target_point)};
+    // Apply the affine map to the points
+    for (size_t d = 0; d < Dim; ++d) {
+      source_point.get(d) =
+          source_point.get(d) * gsl::at(map_inverse_slope_, d) +
+          gsl::at(map_inverse_offset_, d);
+    }
+    return source_point;
+  }
+
+  template <typename T>
+  Tensor<T, tmpl::integral_list<std::int32_t, 2, 1>,
+         index_list<SpatialIndex<Dim, UpLo::Up, Frame::Logical>,
+                    SpatialIndex<Dim, UpLo::Lo, TargetFrame>>>
+  inv_jacobian(tnsr::I<T, Dim, Frame::Logical> source_point) const noexcept {
+    for (size_t d = 0; d < Dim; ++d) {
+      source_point.get(d) = source_point.get(d) * gsl::at(map_slope_, d) +
+                            gsl::at(map_offset_, d);
+    }
+    auto inv_jac = block_map_->inv_jacobian(source_point);
+    for (size_t d = 0; d < Dim; ++d) {
+      for (size_t i = 0; i < Dim; ++i) {
+        inv_jac.get(d, i) *= gsl::at(inverse_jacobian_, d);
+      }
+    }
+    return inv_jac;
+  }
+
+  template <typename T>
+  Tensor<T, tmpl::integral_list<std::int32_t, 2, 1>,
+         index_list<SpatialIndex<Dim, UpLo::Up, TargetFrame>,
+                    SpatialIndex<Dim, UpLo::Lo, Frame::Logical>>>
+  jacobian(tnsr::I<T, Dim, Frame::Logical> source_point) const noexcept {
+    for (size_t d = 0; d < Dim; ++d) {
+      source_point.get(d) = source_point.get(d) * gsl::at(map_slope_, d) +
+                            gsl::at(map_offset_, d);
+    }
+    auto jac = block_map_->jacobian(source_point);
+    for (size_t d = 0; d < Dim; ++d) {
+      for (size_t i = 0; i < Dim; ++i) {
+        jac.get(i, d) *= gsl::at(jacobian_, d);
+      }
+    }
+    return jac;
+  }
+
+  // clang-tidy: do not use references
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+ private:
+  std::unique_ptr<CoordinateMapBase<Frame::Logical, TargetFrame, Dim>>
+      block_map_{nullptr};
+  ElementId<Dim> element_id_{};
+  // length_of_range_[i] = segment_ids[i].endpoint(Side::Upper) -
+  //                       segment_ids[i].endpoint(Side::Lower)
+  std::array<double, Dim> length_of_range_{
+      make_array<Dim>(std::numeric_limits<double>::signaling_NaN())};
+  // length_of_range_[i] = segment_ids[i].endpoint(Side::Upper) +
```

Comment:
It is not clear to me what these comments have to do with the code surrounding them. Additionally, this one appears to contradict the previous one.

Reply:
The reason I added the comments was so it was clearer what the variable below the comment corresponds to (clearly wasn't successful :) ). However, since it is the same as in the constructor I would say it's better to delete them. Do you agree?

Reply:
I kind of like the idea, but even knowing what they are supposed to do I still can't decipher them.  What is `A_`?  What is `Dim.0`?  And I think the one on this line is referring to the wrong variable.

I also wouldn't object if you'd prefer to remove them and just let people look at the constructor since the calculations aren't very complicated in this case.